### PR TITLE
DCOS-22058: change package repositories ADD to the new data-layer patterns

### PR DESCRIPTION
**Pull request by Fabs**
~Depends on #2870, and then we can switch base to `feature/data-layer`. This was a split of that one to make it simpler.~

Moves add repositories away from mixings and flux stores by using the new data-layer patterns (streams and mediators). This is also done in a preparation to use graphql.

TESTING
Just open the Settings > Repositories and try to add, remove and filter packages.
 
AUTOMATED TESTING
No new tests were added since this is a refactor and rely on the current tests.

 **Checklist**
- [x] Explain how to test it
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?

### CODE REVIEW

**Comment by nLight** [comment]
I'm afraid you need to rebase the whole chain to make the NodesPage test go away.

File: plugins/catalog/src/js/repositories/RepositoriesAdd.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+// Imported from the Cosmos Store
+const getErrorMessage = (response = {}) => {
+  if (typeof response === "string") {
+    return response;
+  }
+
+  return response.description || response.message || "An error has occurred.";
+};
+
+const addRepositoryEven$ = new Subject();
```
Comment (ahoskins) [inline plugins/catalog/src/js/repositories/RepositoriesAdd.js]: should this say `addRepositoryEvent$`? Not sure what "even" means.

Comment (Fabs) [reply plugins/catalog/src/js/repositories/RepositoriesAdd.js]: you are right, going to fix that typo.

**Comment by Fabs** [comment]
@nlight. Done.

**Pull request merged**